ui/src/message_popup/author/questions_page.jsx

```diff
@@ -29,12 +30,18 @@ export default React.createClass({
 
   getInitialState(){
     return ({
-      questions: allQuestions,
+      allDatabaseQuestions: [],
+      currentQuestions: [],
+      archivedQuestions: [],
```

Comment:
The reasoning behind separating it out in the state like above is because `allDatabaseQuestions` would remain constant throughout the component's lifecycle, (only changing when`onQuestionsChanged` would be called) but the `currentQuestions` and `archivedQuestions` would add and remove questions depending on search bar input. The full list of questions however, found in `allDatabaseQuestions`, would be needed to restore the `currentQuestions` and `archivedQuestions` lists after the search bar changes again or is cleared.

I now see though from your comment on getting these search bar lists in the render method that I can definitely simplify this.